Artroom1/AppDelegate.swift

```diff
@@ -14,10 +14,45 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-
+      
+        func loadJson(_ fileName: String) -> [Artwork]? {
```

Comment:
Remember when we talked about loading the data etc. and I mentioned you could also move the data loading part into it's own type? I think it's a good time to do that now, so your AppDelegate (this file here) will not get cluttered. In programming there is something called the Single Responsibility Principle. What this means is that any type in your application (class, function etc.) should be focussed on exactly one responsibility.
In this case, as we're in the scope of the `AppDelegate`, loading of data is probably not the responsibility of the delegate. So we could move the `loadJson` function into it's own type (e.g. `DataLoader`). You could then create an instance of the data loader here and call `dataLoader.loadJson()`.

Reply:
Another thing I noticed, you are passing a `fileName`, but you are actually not using it in `Bundle.main.path(forResource: ...)` 🙂

Reply:
You can also already read up on Singletons. [Here's a quite good and easy introduction](https://cocoacasts.com/what-is-a-singleton-and-how-to-create-one-in-swift) (at least I thought so while skimming over it).

Reply:
Coming in the next branch!